docs/articles/advanced_topics/buttons.md

```diff
@@ -0,0 +1,168 @@
+    .WithContent("This message has buttons! Pretty neat innit?")
+    .WithComponents(myButton);
+```
+Now you have a message with a button. Congratulations! It's important to note that `.WithComponents()` will create a new row with each call, so **add everything you want on one row in one call!**
+
+Buttons can be added in any order you fancy. Lets add 5 to demonstrate each color, and a link button for good measure.
+
+```cs
+var builder = new DiscordMessageBuilder()
+    .WithContent("This message has buttons! Pretty neat innit?")
+    .WithComponents(new DiscordComponent[]
+    {
+        new DiscordButtonComponent(ButtonStyle.Primary, "1_top" "Blurple!"),
+        new DiscordButtonComponent(ButtonStyle.Secondary, "2_top", "Grey!"),
+        new DiscordButtonComponent(ButtonStyle.Success, "3_top", "Green!"),
+        new DiscordButtonComponent(ButtonStyle.Danger, "4_top", "Red!"),
+        new DiscordLinkButtonComponent("https://some-super-cool.site", "Link!")
+    });
+```
+As promised, not too complicated. Links however are `DiscordLinkButtonComponent`, which takes a URL as it's first parameter, and the label. Link buttons can also have an emoji, like regular buttons.
+
+Lets also add a second row of buttons, but disable them, so the user can't push them all willy-nilly.
+
+```cs
+builder.WithComponents(new DiscordComponent[] 
+{
+    new DiscordButtonComponent(ButtonStyle.Primary, "1_top_d" "Blurple!", true),
+    new DiscordButtonComponent(ButtonStyle.Secondary, "2_top_d" "Grey!", true),
+    new DiscordButtonComponent(ButtonStyle.Success, "3_top_d", "Green!", true),
+    new DiscordButtonComponent(ButtonStyle.Danger, "4_top_d" "Red!", true),
+    new DiscordButtonComponent("https://some-super-cool.site", "Link!", true)
+});
+```
+Practically identical, but now with `true` as an extra paremeter. This is the `Disabled` property. 
+
+Produces a message like such: ![Buttons](/images/advanced_topics_buttons_01.png)
+
+Well, that's all neat, but lets say you want to add an emoji. Being able to use any emoji is pretty neat, afterall. That's also very simple!
+
+```cs
+var myButton = new DiscordButtonComponent
+(
+    ButtonStyle.Primary, 
+    "emoji_button", 
+    null, 
+    false, 
+    new DiscordComponentEmoji(595381687026843651)
+);
+```
+And you're done! Simply add that to a builder, and when you send, you'll get a message that has a button with a little Pikachu enjoying a lolipop. Adorable. ![PikaLolipop](/images/advanced_topics_buttons_02.png)
+
+
+# Responding to button presses
+
+When any button is pressed, it will fire the [ComponentInteractionCreated](xref:DSharpPlus.DiscordClient#ComponentInteractionCreated) event.
+
+In the event args, `Id` will be the id of the button you specified. There's also an `Interaction` property, which contains the interaction the event created. It's important to respond to an interaction within 3 seconds, or it will time out. Responding after this period will throw a `NotFoundException`.
```

Comment:
```suggestion
In the event args, `Id` will be the id of the button you specified. There's also an `Interaction` property, which contains the interaction the event created. It's important to respond to an interaction **within 3 seconds**, or it will time out. Responding after this period will throw a `NotFoundException`. You can respond to an interaction and then later edit that response if your code needs to more than 3 seconds to process.
```
I feel this people will gloss over the three second portion and ask why their interaction is failing. Also feel people will ask how to get around the three seconds.

Reply:
You can respond to an interaction with `DefferedMessageUpdate` [...]

Reply:
Sorry, I wasn't aware that took into account the time window.

Reply:
> Sorry, I wasn't aware that took into account the time window.

Read the docs :^)